FE/src/components/EditRecipe.jsx

```diff
@@ -1,8 +1,308 @@
+  const [notesList, setNotesList] = useState([{note: ''}]);
+
+  useEffect(() => {
+   async function getRecipeById() {
+       try {
+           const response = await fetch(`${testAPI}recipes/${recipeId}`, {
+               method: "GET",
+               header: `Bearer ${token}`,
+           });
+           const result = await response.json();
+           const recipe = result.recipe;
+           setRecipeObj(recipe);
+           console.log(recipe.title);
+           setTitle(recipe.title);
+           console.log(title)
+           //setImage(recipe.imgurl); for larry to change later
+           setEstTime(recipe.esttime);
+          
+           function setInitialIngred(array) {
+               let initialArray = [];
+               if (array.length >=1) {
+                   array.map((element) => {
+                       initialArray.push({ingredient: element})
+                   });
+               }
+  
+               setIngredientList(initialArray);
+           }
+
+
+           function setInitialInstruct(array) {
+               let initialArray = [];
+               if (array) {
+                   array.map((element) => {
+                       initialArray.push({instruction: element})
+                   });
+               }
+               setInstructionList(initialArray);
+           }
+
+
+           function setInitialNotes(array) {
+               let initialArray = [];
+               if (array.length >= 1) {
+                   array.map((element) => {
+                       initialArray.push({note: element})
+                   });
+               } else {
+                    initialArray.push({note: ''})
+               }
+
+
+               setNotesList(initialArray);
+           }
+
+
+           function setInitialTags(array) {
+
+
+               function rearrange(input, field) {
+                   let output = [];
+                   for (let i = 0; i < input.length; i++) {
+                     output.push(input[i][field]);
+                   }
+                   return output;
+               }
+              
+               if(array.length >= 1) {
+                   let newTagsArray = rearrange(array, 'name');
+
+                   let initialArray = [];
+                   if (newTagsArray.length >=1) {
+                       newTagsArray.map((element) => {
+                           initialArray.push({tag: element})
+                       });
+                   }
+                   setTagsList(initialArray);
+               }
+           }
+
+
+           if (Object.keys(recipeObj).length >= 1) {
+               setInitialIngred(recipeObj.ingredients);
+               setInitialTags(recipeObj.tags)
+               setInitialInstruct(recipeObj.procedure);
+               setInitialNotes(recipeObj.notes);
+           }
+
+
+       } catch (error) {
+           console.error(error);
+       }
+   }
+   getRecipeById();
+  }, [title]);
```

Comment:
This line right here is why you can't edit the title. It's constantly being reset. Try doing this with two useEffects. The first gets the ingredient details on page load, the second sets all the ingredients when recipeObj is changed. I think this will solve the problems you are having.